deploying-monitoring-redis-oke.md

```diff
@@ -0,0 +1,257 @@
+```
+
+Run the Python script to generate the data:
+
+```console
+python names.py
+```
+
+This will create a file.csv in the current directory. You can [configure a PersistentVolume](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/) to store and load the data but for the purpose of this exercise, we'll do a quick hack by installing redis in the bastion:
+
+```console
+sudo yum install redis -y
+```
+
+This will allow us to use the redis-cli from the bastion where we have generated/uploaded the file.csv.
+
+On the bastion, get a list of Redis pods:
+
+```console
+kubectl -n redis get pods
+NAME                             READY   STATUS    RESTARTS   AGE                                                                                
+redis-master-0                   1/1     Running   0          156m                                                                               
+redis-metrics-794db76ff7-xmd2q   1/1     Running   0          156m                                                                               
+redis-slave-7fd8b55f7-25w8d      1/1     Running   1          156m                                                                               
+redis-slave-7fd8b55f7-hvhmc      1/1     Running   1          156m                                                                               
+redis-slave-7fd8b55f7-mjq8q      1/1     Running   1          156m
+```
+
+Afterwards, use port-forward to so you can access it using the redis-cli:
+
+<!--EDIT is there actually a command `k`?-->
+```console
+k -n redis port-forward redis-master-0 6379:6379
+Forwarding from 127.0.0.1:6379 -> 6379
+```
+
+Open a new terminal, login into the bastion and obtain the Redis password:
+
+```console
+export REDIS_PASSWORD=$(kubectl get secret --namespace redis redis -o jsonpath="{.data.redis-password}" | base64 --decode)
+```
+
+Do a quick test to see you can connect to Redis:
+
+<!--EDIT command appears mangled-->
+```console
+redis-cli -a $REDIS_PASSWORD127.0.0.1:6379> ping                                                                                                                                                        
+PONG                                                                                                                                                                        
+127.0.0.1:6379>
+```
+
+Before we import the csv, access Grafana (http://localhost:3000) as described above by opening a third terminal and running kubectl port-forward locally. Browse to the Redis Dashboard and set the refresh to every 5 seconds:
+
+```console
+kubectl -n monitoring port-forward prom-operator-grafana-77cdf86d94-m8pv5 3000:3000
+```
+
+Now import the csv file as follows:
+
+<!--EDIT appears mangled, assume line break in --pipe\nAll, and no line break before file.csv-->
```

Comment:
I went ahead and modified this one so it made sense to me, but please double check it against source.

Reply:
My new commit matches formatting form the source